feature/#508 change Pipeline and PipelineConfig documentation (#535)
Avaiga/taipy-doc: docs/manuals/core/concepts/index.md

Taipy Core is an application builder designed to help Python developers turn their data algorithms into an interactive
production-ready data-driven application. Taipy Core provides the necessary concepts for modeling, executing, and
monitoring algorithms. In this documentation, such concepts are called scenarios.

A scenario can be seen as a succession of functions that exchange data. It can be described as an

> execution graph (a Directed Acyclic Graph or DAG). With Taipy Core, one can model simple as well as very

Reviewer: ```suggestion
execution graph (a Directed Acyclic Graph or DAG). With Taipy Core, one can model simple and very
```

- A [Task](task.md) (the orange boxes) can be seen as a function receiving data node(s) as input and returning
  data node(s) as output.
- A [Job](job.md) represents a unique execution of a Task.

> - A [Scenario](scenario.md) represents a set of tasks connected together through data nodes forming a Directed Acyclic Graph,
>   that should be executed as a whole (or as subgraphs via the use of sequences) and forming a consistent algorithm. It can
>   also contain a set of additional data nodes for data that are related but are not part of the executable graph.

Reviewer: ```suggestion
- A [Scenario](scenario.md) represents a set of tasks connected through data nodes forming a Directed Acyclic Graph
  that should be executed as a whole (or as subgraphs via sequences) and create a consistent algorithm. It can
  also contain a set of additional data nodes for related data that are not part of the executable graph.
```

> - A [Sequence](sequence.md) represents a set of tasks connected together through data nodes, that should be executed

Reviewer: ```suggestion
- A [Sequence](sequence.md) represents a set of tasks connected through data nodes, that should be executed
```
